collections/_article/dev-snapshot-godot-4-2-beta-1.md

```diff
@@ -0,0 +1,382 @@
+
+Please note, that the aforementioned changes to GPU particles mark the point where they start to deviate from CPU particles. This means that in the future more GPU-only features are going to be implemented, leaving CPU particles as a low-cost, simplified alternative, and not a direct 1-to-1 equivalent. GPU particles are supported by all rendering backends in Godot 4, and we encourage users to move towards using them. To aid in that transition, there is now a built-in tool to convert CPU particles to their GPU equivalent ([GH-80779](https://github.com/godotengine/godot/pull/80779) by [Yuri Rubinsky](https://github.com/Chaosus)).
+
+#### Shaders
+
+Visual shaders got some nice improvements with the addition of drop-down list properties to custom nodes ([GH-81688](https://github.com/godotengine/godot/pull/81688) by [Yuri Rubinsky](https://github.com/Chaosus)), and output ports for vector types are now expandable by default ([GH-82088](https://github.com/godotengine/godot/pull/82088) by [Dennis Manaa](https://github.com/DennisManaa), their first contribution).
+
+In text shaders, improvements were done to uniform sampler arrays to give them support for more hint types, such as filter modes ([GH-79100](https://github.com/godotengine/godot/pull/79100) by [Tomasz Chabora](https://github.com/KoBeWi)).
+
+
+### Scripting
+
+Godot has two main scripting languages, GDScript and C#, and each of them deserves special care and attention. That said, the new platform support coming to 4.2 is a very enticing reason to upgrade for C# users specifically!
+
+But let's start with a more general change. The script debugger now comes with full support for threaded code ([GH-76582](https://github.com/godotengine/godot/pull/76582) by [Juan Linietsky](https://github.com/reduz)), including the execution stack and breakpoints. This is something that has been requested since Godot 3 days, and now with the better multithreading support of Godot 4 it is finally possible to debug extra threads.
+
+<video autoplay loop muted playsinline title="A demo of new threading capabilities in the script debugger">
+  <source src="/storage/blog/godot-4-2-beta/threaded-debugger.mp4" type="video/mp4">
+</video>
+
+#### C# / .NET
+
+Back in March, Godot 4.0 was released replacing Mono support in favor of the new .NET framework, which created an unfortunate vacuum when it comes to mobile and web platform support. Godot C#/.NET maintainers had to wait for the new .NET 8.0 to continue their work and expand support to include those platforms. And so the time has come. Godot 4.2 features experimental support for Android (.NET 7.0+ required) and iOS (.NET 8.0 required) export targets.
```

Comment:
I'm a bit worried that users may misunderstand this part because:
- Saying "replacing Mono support" may give users the impression that we are not using Mono at all. We did get rid of the Mono embedding APIs, but we may still use the Mono runtime when appropriate (mobile and web platforms).
- The name ".NET framework" may be confused with ".NET Framework" (noticed the uppercase **F**) which refers to the old Windows-only implementation. Perhaps this part could be replaced with just ".NET"?

Reply:
If we use Mono in future, it will be reintroduced then. I don't think it's incorrect to say that we've replaced it.

Yeah, I am slightly concerned about ".NET framework", but saying "in favor of the new .NET" or just "in favor of .NET" doesn't sound good to me either. Ultimately, even if some would be confused by it, it shouldn't cause any problems. But I can use some other generic term, if you have suggestions (e.g. ".NET library" or something like that).

Reply:
> If we use Mono in future, it will be reintroduced then.

Well, we are using the Mono runtime in our experimental export for Android and iOS, that's why I was saying it. But it's probably fine.

> But I can use some other generic term, if you have suggestions

I mean the word framework is not incorrect, maybe we could say ".NET Core framework" since that's technically what it is (even though Microsoft dropped the "Core" part of the name in .NET 5.0).

Reply:
Yeah, I didn't use Core because they renamed it again. It's a hot mess :D 

> Well, we are using the Mono runtime in our experimental export for Android and iOS, that's why I was saying it. But it's probably fine.

That sentence is about 4.0 though. I can mention that Android support already reintroduces Mono, if you want!

Reply:
I reworked the sentence a bit to address both issues. Will push soon.